### PR TITLE
Default userName after signup

### LINKED ISSUE #2018: Force user to choose username after signup
Found a bug? Please fill out the sections below. 👍

### Issue Summary
I have added new user to the team and he signed up and tried to create a booking for one of the default event but it gave 404 error because of incorrect url it caused because user didn't setup the username.

image
image

### Steps to Reproduce

1. Add new user to team
2. Sign up
3. Open booking and select any event

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Technical details
Occurred on self-hosted version

Solution would be to add the username field while on-boarding or force user to add the username before doing certain actions.

### PR DESCRIPTION
**Pull request by andreaestefania12**
## What does this PR do?

Fixes #2018
When you create a new account on Cal.com and you do not have username, it going to give you one with your name and some random letters. I only can tested create accounts from Prisma Data Platform. 
image

Once the registration is complete you can change the username in settings.
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

1. Add new user to team.
2. Sign up with the new user. (This user has to be created in DB). 
3. Once the registration is complete you can see a random username.
4. Go back to the owner team account and open view public page, booking and select any event
image

### CODE REVIEW

File: apps/web/pages/api/user/[id].ts [hunk 1]
```diff
@@ -4,6 +4,9 @@ import type { NextApiRequest, NextApiResponse } from "next";
 import { getSession } from "@lib/auth";
 import prisma from "@lib/prisma";
 
+import slugify from "@lib/slugify";
+import { randomString } from "@lib/random";
```
Comment (miguelnietoa) [inline apps/web/pages/api/user/[id].ts]:
```suggestion
import { getSession } from "@lib/auth";
import prisma from "@lib/prisma";
import { randomString } from "@lib/random";
import slugify from "@lib/slugify";
```

File: apps/web/pages/api/user/[id].ts [hunk 2]
```diff
@@ -33,6 +36,28 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
+      },
+      select: {
+        username: true,
+        name: true,
+      },
+    });
+
+    if(!user?.username && user?.name && req.body.data?.completedOnboarding){
+      const usernameSlug = (username: string) => slugify(username) + "-" + randomString(6).toLowerCase();
+      const firstUserName =  usernameSlug(user.name);
```
Comment (miguelnietoa) [inline apps/web/pages/api/user/[id].ts]:
```suggestion
    if (!user?.username && user?.name && req.body.data?.completedOnboarding) {
      const usernameSlug = (username: string) => slugify(username) + "-" + randomString(6).toLowerCase();
      const firstUserName = usernameSlug(user.name);
```

File: apps/web/pages/api/user/[id].ts [hunk 3]
```diff
@@ -33,6 +36,28 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   }
 
   if (req.method === "PATCH") {
+    const user = await prisma.user.findFirst({
```
Comment (miguelnietoa) [inline apps/web/pages/api/user/[id].ts]: I think it's better to use `prisma.user.findUnique` since the `id` is unique. What do you think?

**Review by miguelnietoa — CHANGES_REQUESTED** [review]
Looks good to me, @andreaestefania12!

Some changes are required, and I think it is ready to create PR on cal.com official repository.

File: apps/web/pages/api/user/[id].ts (hunk 3 above)
Comment (andreaestefania12) [reply apps/web/pages/api/user/[id].ts]: You're right, I'll change it

**Pull request closed without merging**